lucetc/tests/wasi-sdk.rs

```diff
@@ -87,7 +87,7 @@ mod programs {
         let b = Bindings::empty();
         let h = HeapSettings::default();
         let c = Compiler::new(&m, OptLevel::Best, &b, h).expect("compile a");
-        let _mdata = c.module_data();
+        let _mdata = c.module_data().unwrap();
 
         /* FIXME: module data doesn't contain the information to check these properties:
         assert_eq!(p.import_functions().len(), 0, "import functions");
```

Comment:
@pchickey can you think of a reason we'd discerning between import and export functions in a list of names/signatures/etc? if not, I'll uncomment relevant parts of the lucetc tests too

Reply:
As future work, we do want some way to check the set of import functions and their signatures for a module, as a runtime safety feature that e.g. checks that all of the wasi imports match the types that the wasi runtime library implemented them as.  We will want to do the same in the compiler, but since deployment situations will doubtless be complicated, I would like to check them in both places since the check will be quick & easy on module load time.

The tests happened to be written this way against the old abstraction because its a reasonable way to check that a given wat's imports and exports match up against the compiled version. It might not end up making sense in the wasi-sdk test suite but it would be nice to try it. If it seems fragile or fiddly in this particular test, we can ignore it.

Reply:
Oh, we will also want signatures on import functions for when we implement per-instance dynamic imports.